vllm/compilation/backends.py

```diff
@@ -55,9 +55,14 @@ def make_compiler(compilation_config: CompilationConfig) -> CompilerInterface:
             logger.debug("Using InductorAdaptor")
             return InductorAdaptor()
     else:
-        assert compilation_config.backend == "eager", (
-            "Custom backends not supported with CompilationMode.VLLM_COMPILE"
-        )
+        if current_platform.is_tpu():
```

Comment:
We should never reach this code unless `CompilationConfig.mode== 3` (`CompilationMode.VLLM_COMPILE`)

Does this mean we're using `VLLM_COMPILE` for TPU now? I thought we used `DYNAMO_TRACE_ONCE`?